### PR TITLE
12 mobx mars viewer

### PR DESCRIPTION
**Pull request by user-of-github**

### CODE REVIEW

**Comment by dex157** [comment]
Mergr request should contain only relevant changes. This is mars viewer homework MR, but it contains a lot of changes regarding other homework. This is bad.

**Comment by user-of-github** [comment]
> Mergr request should contain only relevant changes. This is mars viewer homework MR, but it contains a lot of changes regarding other homework. This is bad.

But I haven't done any _merge requests_. I even have no idea what is it. And if you mean my branches checkouts, than I was just doing everything according to mentor's tutorial: just created every time a new branch for a new homework (via _git checkout -b_ )

And what about directly my project and code ?

File: homeworks/12-mobx-mars/src/components/header.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import Style from './header.module.css'
+
+export const Header = (): JSX.Element => (
+    <header className={Style.header}>
+        <div className="container">
+            <div className={Style.header__mainTitles}>
+                <h1 className={Style.header__title}>Mars Viewer: MobX EDITION</h1>
```
Comment (dex157) [inline homeworks/12-mobx-mars/src/components/header.tsx:8]: If you use `CSS-modules` you could use just `title`, `subtitle`, etc, without the prefix `header`. Webpack will add hash-sum to the class to ignore name collisions.

File: homeworks/12-mobx-mars/src/store/marsViewerState.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import {makeAutoObservable} from 'mobx'
+import {IMarsViewerState, PhotoType, Tab} from '../types/types'
```
Comment (dex157) [inline homeworks/12-mobx-mars/src/store/marsViewerState.ts:2]: Why you don't need to use I prefix https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript/41967120#41967120

File: homeworks/12-mobx-mars/src/store/marsViewerState.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    public loaded: boolean = false
+    public loading: boolean = false
+
+    public constructor(defaultSolution: number) {
+        this.currentSolution = defaultSolution
+        makeAutoObservable(this, {}, {deep: true})
+    }
+
+    public loadNewSolution(): void {
+        !this.loading && fetchPhotos(this)
```
Comment (dex157) [inline homeworks/12-mobx-mars/src/store/marsViewerState.ts:21]: This is not a clear practic pass model to the utility function. The function `fetch` knows a lot about the details of the implementation of the `MarsViewerState` class. Now, if you change the details of your model, you will need to change the implementation of the utility function.

**Review by dex157 — COMMENTED** [review]
Excellent UI! 
The loader from prev HW looks very good, and I'm glad to see clear UI and layout.

**Pull request merged**